### PR TITLE
Replace `AlertDialog` by `showModalBottomSheet`

### PR DESCRIPTION
**Pull request by ggirotto**
It also adds a new color swatch named `negative`

### CODE REVIEW

File: lib/application/theme/memo_theme_colors.dart [hunk 1]
```diff
@@ -46,6 +46,25 @@ MaterialColor buildClassicSecondarySwatch() {
   );
 }
 
+MaterialColor buildClassicNegativeSwatch() {
```
Comment (ggirotto) [inline lib/application/theme/memo_theme_colors.dart:49]: I also like `danger` or `destructive`. Open to suggestions

Comment (matuella) [reply lib/application/theme/memo_theme_colors.dart:49]: I do prefer destructive - totally personal preference, would be good with negative as well.

Danger is too specific

File: lib/application/widgets/theme/negative_button.dart [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:memo/application/theme/theme_controller.dart';
+
+/// Negative alternative to [ElevatedButton].
+///
+/// Used in scenarios where the user action may be destructive
```
Comment (matuella) [inline lib/application/widgets/theme/negative_button.dart:7]:
```suggestion
/// Destructive alternative to [ElevatedButton].
///
/// Used in scenarios that emphasizes an action with impactful consequences.
```

File: lib/application/widgets/theme/negative_button.dart [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+class NegativeButton extends HookWidget {
+  const NegativeButton({required this.onPressed, required this.child, Key? key}) : super(key: key);
```
Comment (matuella) [inline lib/application/widgets/theme/negative_button.dart:9]: Just like I suggested before, I think destructive would be better - but really personal, I'd say negative is good as well.

My only rationale with preferring the former is that destructive is not necessarily a negative behavior - meaning that destroying something is completely acceptable, although requires enough emphasis.

File: lib/application/constants/strings.dart [hunk 4]
```diff
@@ -65,6 +65,8 @@ const executionAnswer = 'Resposta';
 const executionNext = 'Próxima';
 const executionCheckAnswer = 'Ver resposta';
 
+const executionDiscard = 'Descartar';
+const executionBackToStudy = 'Voltar ao estudo';
```
Comment (matuella) [inline lib/application/constants/strings.dart:69]:
```suggestion
const executionBackToStudy = 'Voltar ao Estudo';
```

Following the same as the destructive action text

**Review by matuella — CHANGES_REQUESTED** [review]

File: lib/application/theme/memo_theme_colors.dart (hunk 1 above)
Comment (ggirotto) [reply lib/application/theme/memo_theme_colors.dart]: Updated do destructive

File: lib/application/widgets/theme/negative_button.dart (hunk 3 above)
Comment (ggirotto) [reply lib/application/widgets/theme/negative_button.dart]: Updated do destructive

**Review by matuella — APPROVED** [review]

**Pull request merged**